### PR TITLE
Add LICENSE and NOTICE resources to all modules

### LINKED ISSUE #263: A few more issues with JWT 2.0 release
When I tried to do a fix on JWT master branch for attaching source jar for test dir, I noticed a few more issues:
1. In the past, a few profiles were defined in the tck pom.xml. Now, they were removed. I think the two profiles should be added back in as at least one of them were mentioned in READM.ADOC.
2. README.ADOC needs to be updated to mention JWT 2.0 as the current version mentions 1.2.
3. None of the license files were made to any of the jars in api or tck. I think this related to https://github.com/eclipse/microprofile-parent/issues/50.

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #263

### CODE REVIEW

**Comment by radcortez** [comment]
This shouldn't be needed. Only the Apache License in `resources/META-INF` and a `tck` marker file. The profile in the parent pom will do the replacement.

**Review by Emily-Jiang — COMMENTED** [review]
The changes for TCK are ok. However, you need to add src/main/resources/META-INF/LICENSE and NOTICE for both API and SPEC. Please take a look at MP Config. You can do under this PR or submit a separate one.

**Comment by sberyozkin** [comment]
@radcortez @Emily-Jiang I've aligned it with MP Config, the root folder and `resources/META-INF` for `api/spec/tck` has `LICENSE` and `NOTICE` and `tck/resources/META-INF` already has a `tck` marker

**Comment by radcortez** [comment]
Great. Did you confirm that you get the correct `LICENSE` when you build with `Final`?

File: spec/src/main/META-INF/NOTICE [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+==  NOTICE file corresponding to section 4(d) of the Apache License,   ==
+==  Version 2.0, in this case for Microprofile Config                  ==
+=========================================================================
+
+SPDXVersion: SPDX-2.1
+PackageName: Eclipse Microprofile
+PackageHomePage: http://www.eclipse.org/microprofile
+PackageLicenseDeclared: Apache-2.0
+
+PackageCopyrightText: <text>
```
Comment (Emily-Jiang) [inline spec/src/main/META-INF/NOTICE:11]: Please update the contributor list.

File: tck/src/main/resources/META-INF/NOTICE [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+SPDXVersion: SPDX-2.1
+PackageName: Eclipse Microprofile
+PackageHomePage: http://www.eclipse.org/microprofile
+PackageLicenseDeclared: Apache-2.0
+
+PackageCopyrightText: <text>
+Mark Struberg struberg@apache.org,
```
Comment (Emily-Jiang) [inline tck/src/main/resources/META-INF/NOTICE:15]: same here.

**Comment by sberyozkin** [comment]
@Emily-Jiang Emily, I've remove `PackageCopyrightText` in all `NOTICE` files as I don't know the emails of all the contributors, I think the root `pom` `developers` section is sufficient, I just added @rdebusscher there - I pretty sure he was added back in 1.2 but may be it got lost somehow

**Comment by sberyozkin** [comment]
Let me add @ayoho as he opened a few PRs

**Comment by sberyozkin** [comment]
I've added 3 developers. Several more people have contributed - I'd just like to see 3 or so PRs or some active contributions to the discussions.

File: NOTICE [hunk 3]
```diff
@@ -0,0 +1,13 @@
+=========================================================================
+==  NOTICE file corresponding to section 4(d) of the Apache License,   ==
+==  Version 2.0, in this case for Microprofile Config                  ==
```
Comment (Emily-Jiang) [inline NOTICE:3]:
```suggestion
==  Version 2.0, in this case for MicroProfile JWT Auth                  ==
```

File: NOTICE [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+=========================================================================
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+SPDXVersion: SPDX-2.1
+PackageName: Eclipse Microprofile
```
Comment (Emily-Jiang) [inline NOTICE:10]:
```suggestion
PackageName: MicroProfile
```

File: spec/src/main/META-INF/NOTICE [hunk 5]
```diff
@@ -0,0 +1,10 @@
+=========================================================================
+==  NOTICE file corresponding to section 4(d) of the Apache License,   ==
+==  Version 2.0, in this case for Microprofile Config                  ==
```
Comment (Emily-Jiang) [inline spec/src/main/META-INF/NOTICE:3]:
```suggestion
==  Version 2.0, in this case for Microprofile JWT Auth                  ==
```

File: spec/src/main/META-INF/NOTICE [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+=========================================================================
+
+SPDXVersion: SPDX-2.1
+PackageName: Eclipse Microprofile
```
Comment (Emily-Jiang) [inline spec/src/main/META-INF/NOTICE:7]:
```suggestion
PackageName: MicroProfile
```

File: tck/src/main/resources/META-INF/NOTICE [hunk 7]
```diff
@@ -0,0 +1,13 @@
+=========================================================================
+==  NOTICE file corresponding to section 4(d) of the Apache License,   ==
+==  Version 2.0, in this case for Microprofile Config                  ==
```
Comment (Emily-Jiang) [inline tck/src/main/resources/META-INF/NOTICE:3]:
```suggestion
==  Version 2.0, in this case for Microprofile JWT Auth                  ==
```

File: tck/src/main/resources/META-INF/NOTICE [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+=========================================================================
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+SPDXVersion: SPDX-2.1
+PackageName: Eclipse Microprofile
```
Comment (Emily-Jiang) [inline tck/src/main/resources/META-INF/NOTICE:10]:
```suggestion
PackageName: MicroProfile
```

**Comment by Emily-Jiang** [comment]
@sberyozkin nearly there. I spotted a few more copy and paste errors, so I have made the suggestions.

**Comment by sberyozkin** [comment]
@Emily-Jiang thanks, let me squash the commits

**Comment by sberyozkin** [comment]
Hey @radcortez Yes, with `-Drelease.revision=Final` I can see an EF license being linked to from `tck-javadocs` `index.html` and I guess from other places too

**Comment by sberyozkin** [comment]
@radcortez @Emily-Jiang Looks like the jobs are hanging, not that I've changed any code in this PR

**Review by Emily-Jiang — APPROVED** [review]

**Comment by Emily-Jiang** [comment]
@sberyozkin It looks like all jobs have finally completed. Thank you for getting this done! Finally ready for RC2, Yay...

**Comment by sberyozkin** [comment]
@Emily-Jiang thanks

**Pull request merged**